### PR TITLE
Expiration controls and custom logger

### PR DESCRIPTION
**Pull request by nwithan8**
Set expiration time for interactions (how long since it was recorded should an interaction be considered valid)
  - Can determine what to do if a matching interaction is considered invalid:
    - Warn the user, but proceed with the interaction
    - Throw an exception
    - Automatically re-record (cannot use this in `Replay` mode)

Users can pass in a logger to EasyVCR (will be used to log warnings for expiration; otherwise, log to console)

### CODE REVIEW

File: src/main/java/com/easypost/easyvcr/ExpirationActions.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ */
+public enum ExpirationActions {
+    /**
+     * Warn that the recorded interaction is expired, but proceed as normal.
+     */
+    Warn,
+    /**
+     * Throw an exception that the recorded interaction is expired.
+     */
+    Throw_Exception,
```
Comment (Justintime50) [inline src/main/java/com/easypost/easyvcr/ExpirationActions.java:14]: Question: The title case split by an underscore is a Java convention?...

File: src/main/java/com/easypost/easyvcr/clients/httpurlconnection/RecordableHttpURLConnection.java [hunk 2]
```diff
@@ -265,6 +318,7 @@ private void buildCache() throws VCRException {
                         cacheInteraction(true);
                     }
                 } catch (VCRException | InterruptedException e) {
+                    // does not catch RecordingExpirationException, so that exception will make it through
```
Comment (Justintime50) [inline src/main/java/com/easypost/easyvcr/clients/httpurlconnection/RecordableHttpURLConnection.java:321]: Thought: great use of a comment

File: src/main/java/com/easypost/easyvcr/internalutilities/ConsoleFallbackLogger.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package com.easypost.easyvcr.internalutilities;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * A logger that, if there's no proper logger, writes to the console instead.
```
Comment (Justintime50) [inline src/main/java/com/easypost/easyvcr/internalutilities/ConsoleFallbackLogger.java:7]: Nit: I'd clarify this maybe? If there is no `custom` or `overridden` logger, right?

**Review by Justintime50 — APPROVED** [review]
This was an ideal PR, nice work 🙂

**Comment by nwithan8** [comment]
> This was an ideal PR, nice work 🙂

thanks-pops-david-james-elliott

File: src/main/java/com/easypost/easyvcr/ExpirationActions.java (hunk 1 above)
Comment (nwithan8) [reply src/main/java/com/easypost/easyvcr/ExpirationActions.java:14]: Convention is all caps, underlines for spaces: https://howtodoinjava.com/java/enum/enum-tutorial/#:~:text=By%20convention%2C%20enums%20are%20constants,(same%20as%20class%20names).

But I noticed that, oops, all the other enums in here are title cased like the C# version.

And then I just gave up at this weird hybrid...

Comment (nwithan8) [reply src/main/java/com/easypost/easyvcr/ExpirationActions.java]: We can put down enum renaming as a breaking change for the future.

For now, the new enums with match the TitleCase format of the others.

**Review by jchen293 — APPROVED** [review]
You can have a helper function for the nested switch, but won't block it

File: src/main/java/com/easypost/easyvcr/ExpirationActions.java (hunk 1 above)
Comment (Justintime50) [reply src/main/java/com/easypost/easyvcr/ExpirationActions.java]: The beauty of a pre v1 release is we can break all we want. Let's do so separately, but let's also do this prior to going stable. Can you make a task to follow up?

**Comment by nwithan8** [comment]
> You can have a helper function for the nested switch, but won't block it

Backlog 😅

**Pull request merged**